tests/src/staking.test.ts

```diff
@@ -0,0 +1,561 @@
+import * as anchor from "@project-serum/anchor";
+import { Transaction } from "@raindrop-studios/sol-kit";
+import {
+  ItemProgram,
+  StakingProgram,
+  Instructions,
+  Utils,
+} from "@raindrops-protocol/raindrops";
+import { IDL as ItemProgramIDL } from "./types/item";
+import { IDL as StakingProgramIDL } from "./types/staking";
+import {
+  createMintNFTInstructions,
+  createMintTokensInstructions,
+} from "./utils/token";
+
+describe("Staking Program", () => {
+  it.concurrent("Item Staking", async () => {
+    const walletKeypair = anchor.web3.Keypair.generate();
+
+    const provider = new anchor.AnchorProvider(
+      new anchor.web3.Connection("http://localhost:8899"),
+      new anchor.Wallet(walletKeypair),
+      { commitment: "confirmed" }
+    );
+
+    await provider.connection.requestAirdrop(
+      walletKeypair.publicKey,
+      100 * anchor.web3.LAMPORTS_PER_SOL
+    );
+
+    const itemProgram = await ItemProgram.getProgramWithConfig(ItemProgram, {
+      asyncSigning: false,
+      provider,
+      idl: ItemProgramIDL,
+    });
+
+    const stakingProgram = await StakingProgram.getProgramWithConfig(
+      StakingProgram,
+      {
+        asyncSigning: false,
+        provider,
+        idl: StakingProgramIDL,
+      }
+    );
+
+    let ixs, args, accounts, additionalArgs;
+
+    // STEP 1: Creates an item class
```

Comment:
We need to wrap these steps out into their own functions or have a factory of some sort that gives a way to list what we want built before we can run our tests. Something to the effect of, I want to test "STEP 8: Begins artifact stake warmup", factory, please create me state that does `[STEP-1, STEP-2, ..., STEP-7]`, so I can just test 8 in isolation and not have these huge test cases for the entire flow.